### PR TITLE
generate bibliography

### LINKED ISSUE #75: Migrate bibliography
https://openproblems-experimental.netlify.app/bibliography.html is empty

### PR DESCRIPTION
**Pull request by KaiWaldrant**
closes #75

### CODE REVIEW

**Review by rcannood — CHANGES_REQUESTED** [review]
Wow, this works? Nice find :)

* [`/bibliography`](https://issue-75-bibliography--openproblems-experimental.netlify.app/bibliography/) is empty but [`/bibliography/index.html`](https://issue-75-bibliography--openproblems-experimental.netlify.app/bibliography/index.html) is not. Can you remove `/bibliography.qmd`?
* Can you reuse `/library.bib` instead of committing another `/bibliography/main.bib`?

**Comment by scottgigante** [comment]
> Wow, this works? Nice find :)

Apparently not (: 

image

**Comment by KaiWaldrant** [comment]
locally it works but for some reason not rendering on the netlify page
image

**Comment by rcannood** [comment]
Should be fixed now :) There was a `/bibliography.qmd` file and `/bibliography/index.qmd` file in the repo.

**Comment by scottgigante** [comment]
The # refs are not quite clear enough imo. See the existing page on the original website -- I use a `::before` icon, a background colour and an animation to make clear which reference was clicked.

**Comment by scottgigante** [comment]
Also `@misc` references are not rendered properly. See e.g.
```
———. 2019. “5k Peripheral Blood Mononuclear Cells (PBMCs) from a Healthy Donor with a Panel of TotalSeq-b Antibodies (V3 Chemistry).” https://www.10xgenomics.com/resources/datasets/5-k-peripheral-blood-mononuclear-cells-pbm-cs-from-a-healthy-donor-with-cell-surface-proteins-v-3-chemistry-3-1-standard-3-1-0.
```

**Comment by KaiWaldrant** [comment]
> Also `@misc` references are not rendered properly. See e.g.
> 
> > ———. 2019. “5k Peripheral Blood Mononuclear Cells (PBMCs) from a Healthy Donor with a Panel of TotalSeq-b Antibodies (V3 Chemistry).” https://www.10xgenomics.com/resources/datasets/5-k-peripheral-blood-mononuclear-cells-pbm-cs-from-a-healthy-donor-with-cell-surface-proteins-v-3-chemistry-3-1-standard-3-1-0.

rendering was correct but because quarto uses pandoc to style the biblio, the chicago manual of style was used by default ([here](https://quarto.org/docs/authoring/footnotes-and-citations.html#citation-style)). I changed the csl to human-nature.

File: bibliography/biblio.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.csl-entry:target::before {
+    font: 70% sans-serif;
+    content: "►";
+    margin-right: 0.25em;
+}
+
+
+.csl-entry:target {
+    -webkit-animation: target-fade 2s;
```
Comment (scottgigante) [inline bibliography/biblio.css:9]: You have not defined `target-fade` at all. From the hugo website:

```
@-webkit-keyframes target-fade {
	from { background-color: #ff414b; }
	to { background-color: #ff414b44; }
}
@-moz-keyframes target-fade {
	from { background-color: #ff414b; }
	to { background-color: #ff414b44; }
}
@-o-keyframes target-fade {
	from { background-color: #ff414b; }
	to { background-color: #ff414b44; }
}
@keyframes target-fade {
	from { background-color: #ff414b; }
	to { background-color: #ff414b44; }
}
```

**Review by scottgigante — CHANGES_REQUESTED** [review]
Looks like some more rendering issues that were not present in the old bibliography.
> Schober, P., Boer, C., & Schwarte, L. A. (2018). Correlation coefficients. Anesthesia ***&amp\mathsemicolon***
 Analgesia, 126(5), 1763–1768. https://doi.org/10.1213/ane.0000000000002864

> Venna, J., & Kaski, S. (2001). Neighborhood preservation in nonlinear projection methods: An experimental study. In Artificial neural networks ICANN 2001 (pp. 485–491). Springer Berlin Heidelberg. [https://doi.org/***{10.1007/3-540-44668-0\_68}***](https://doi.org/%7B10.1007/3-540-44668-0/_68%7D)

**Comment by KaiWaldrant** [comment]
For some reason quarto doesn't translate all the latex commands in the `.bib` file  I have changed the ones that were possible. For the others I started a discussion  quarto-dev/quarto-cli#4662

**Comment by rcannood** [comment]
It's weird that the Latex commands inside the bib file aren't being rendered properly. Good idea to create a post for it. It may help to create a small repository which allows the maintainers to simply run `quarto preview` to try to replicate the issue.

@KaiWaldrant Can you check whether the issue persists when using quarto 1.3?

An quick solution might be to add [this shortcode](https://github.com/openproblems-bio/website/blob/main/layouts/shortcodes/bibliography.html) as a html block in the `bibliography/index.qmd`.

File: bibliography/biblio.css [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    -moz-animation: target-fade 2s;
+    -o-animation: target-fade 2s;
+    animation: target-fade 2s;
+    background-color: #ff414b44;
+}
+
+
+[@-webkit-keyframes](https://github.com/-webkit-keyframes) target-fade {
```
Comment (scottgigante-immunai) [inline bibliography/biblio.css:17]: Is this valid CSS?

File: bibliography/library.bib [hunk 3]
```diff
@@ -893,11 +893,11 @@ @inproceedings{venna2001neighborhood
   title         = {Neighborhood Preservation in Nonlinear Projection Methods: An Experimental Study},
   author        = {Jarkko Venna and Samuel Kaski},
   year          = {2001},
-  booktitle     = {Artificial Neural Networks {\textemdash} {ICANN} 2001},
+  booktitle     = {Artificial Neural Networks {---} {ICANN} 2001},
   publisher     = {Springer Berlin Heidelberg},
   pages         = {485--491},
-  doi           = {{10.1007/3-540-44668-0\_68}},
-  url           = {{https://doi.org/10.1007/3-540-44668-0\_68}},
+  doi           = {10.1007/3-540-44668-0_68},
```
Comment (scottgigante-immunai) [inline bibliography/library.bib:899]: Problem here is this is no longer valid bibtex...

Comment (KaiWaldrant) [reply bibliography/library.bib:899]: that's what I guessed

File: bibliography/biblio.css (hunk 2 above)
Comment (KaiWaldrant) [reply bibliography/biblio.css:17]: no, don't know how that got in there

**Review by scottgigante-immunai — CHANGES_REQUESTED** [review]
@KaiWaldrant doesn't look like your css selectors find the bib entries now.

**Review by scottgigante-immunai — APPROVED** [review]

**Pull request merged**